swagger.yaml

```diff
@@ -0,0 +1,221 @@
+swagger: "2.0"
+info:
+  description: "An API used to hold the state of a customised dataset job. Once the job has been created via the `Dataset API` it can then be updated with filters to apply to the dataset. Once the job is ready it is expected that the client will update the job state to start the processing. Once processing has been started the job will be placed in a locked condition as no amendments can be done."
+  version: "1.0.0"
+  title: "Filter API"
+  license:
+    name: "Open Government Licence v3.0"
+    url: "http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
+basePath: "/v1"
+tags:
+- name: "Jobs"
+  description: "Used for storing the state of a customised dataset and retrieving the results"
+schemes:
+- "http"
+parameters:
+  jobId:
+    name: jobId
+    type: string
+    required: true
+    description: "The unique jobId for customising a dataset"
+    in: path
+  dimension:
+    name: dimension
+    schema:
+      $ref: '#/definitions/Dimension'
+    required: true
+    description: "A dimension to filter the dataset"
+    in: body
+  state:
+    name: state
+    schema:
+      $ref: '#/definitions/State'
+    required: true
+    description: "The requested state of the job"
+    in: body
+  newJob:
+    name: newJob
+    schema:
+      $ref: '#/definitions/NewJob'
+    required: true
+    description: "A new job to create"
+    in: body
+paths:
+  /job:
+    post:
+      tags:
+      - "Jobs"
+      summary: "Create a filter job for a dataset"
+      description: "Create a job so that dimensions can be added to filter a dataset"
+      produces:
+      - "application/json"
+      parameters:
+      - $ref: '#/parameters/newJob'
+      responses:
+        200:
+          description: "Job was created"
+          schema:
+            $ref: '#/definitions/JobInstance'
+        500:
+          description: "Failed to process the request due to an internal error"
+  /jobs/{jobId}:
+    get:
+      tags:
+      - "Jobs"
+      summary: "Get a description of a created job"
+      description: "A jobId is required to access a customised dataset job. To create a jobId see the `Dataset API`"
+      produces:
+      - "application/json"
+      parameters:
+      - $ref: '#/parameters/jobId'
+      responses:
+        200:
+          description: "The job was found and a `JobState` is returned"
+          schema:
+            $ref: '#/definitions/JobState'
+        404:
+          description: "Job not found"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /jobs/{jobId}/filter:
```

Comment:
We will need a way to delete existing filters. This may open up a few considerations like how to reference existing filters. We can discuss this further when your back in next week.